src/status_im/contexts/communities/overview/view.cljs

```diff
@@ -137,119 +109,112 @@
+                      {:number-of-hold-tokens number-of-hold-tokens})
+          (i18n/label :t/you-must-hold))])
+     [quo/token-requirement-list
+      {:tokens   tokens
+       :padding? true}]
+     [quo/button
+      {:on-press            #(rf/dispatch [:open-modal :community-account-selection
+                                           {:community-id id}])
+       :accessibility-label :join-community-button
+       :customization-color color
+       :container-style     {:margin-horizontal 12 :margin-top 12 :margin-bottom 12}
+       :disabled?           (not can-request-access?)
+       :icon-left           (if can-request-access? :i/unlocked :i/locked)}
+      (i18n/label :t/join-open-community)]]))
 
-(defn join-community
-  [{:keys [joined color permissions token-permissions membership-permissions? id] :as community}
-   pending?]
-  (let [access-type     (get-access-type (:access permissions))
+(defn- join-community
+  [{:keys [id color joined permissions role-permissions? can-join?] :as community}]
+  (let [pending?        (rf/sub [:communities/my-pending-request-to-join id])
+        access-type     (get-access-type (:access permissions))
         unknown-access? (= access-type :unknown-access)
         invite-only?    (= access-type :invite-only)]
     [:<>
      (when-not (or joined pending? invite-only? unknown-access?)
-       (if membership-permissions?
-         [token-gates community]
+       (if role-permissions?
+         [token-requirements community]
          [quo/button
-          {:on-press
-           (if config/community-accounts-selection-enabled?
-             #(rf/dispatch [:open-modal :community-account-selection {:community-id id}])
-             #(rf/dispatch [:open-modal :community-requests-to-join community]))
-
+          {:on-press            #(rf/dispatch [:open-modal :community-account-selection
+                                               {:community-id id}])
            :accessibility-label :show-request-to-join-screen-button
            :customization-color color
-           :icon-left :i/communities}
+           :icon-left           :i/communities}
           (i18n/label :t/request-to-join)]))
-
-     (when (not (or joined pending? token-permissions))
+     (when (not (or pending? role-permissions? can-join?))
        [quo/text
         {:size   :paragraph-2
          :weight :regular
          :style  style/review-notice}
         (i18n/label :t/community-admins-will-review-your-request)])]))
 
-(defn status-tag
-  [pending? joined]
-  (when (or pending? joined)
-    [rn/view {:style {:position :absolute :top 12 :right 12}}
-     [quo/status-tag
-      {:status {:type (if joined :positive :pending)}
-       :label  (if joined
-                 (i18n/label :t/joined)
-                 (i18n/label :t/pending))}]]))
+(defn- status-tag
+  [community-id joined]
+  (let [pending? (rf/sub [:communities/my-pending-request-to-join community-id])]
```

Comment:
FYI: I moved `pending?` down the tree of components instead of having it at the top `view` component to avoid having to re-render everything.